lightning-background-processor/src/fwd_batch.rs

```diff
@@ -0,0 +1,486 @@
+// This file is Copyright its original authors, visible in version control
+// history.
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+use core::sync::atomic::{AtomicU16, Ordering};
+use core::time::Duration;
+
+pub(crate) struct BatchDelay {
+	next_batch_delay_millis: AtomicU16,
+}
+
+impl BatchDelay {
+	pub(crate) fn new() -> Self {
+		let next_batch_delay_millis = AtomicU16::new(rand_batch_delay_millis());
+		Self { next_batch_delay_millis }
+	}
+
+	pub(crate) fn get(&self) -> Duration {
+		Duration::from_millis(self.next_batch_delay_millis.load(Ordering::Acquire) as u64)
+	}
+
+	pub(crate) fn next(&self) -> Duration {
+		let next = rand_batch_delay_millis();
+		self.next_batch_delay_millis.store(next, Ordering::Release);
+		Duration::from_millis(next as u64)
+	}
+}
+
+fn rand_batch_delay_millis() -> u16 {
+	const USIZE_LEN: usize = core::mem::size_of::<usize>();
+	let mut random_bytes = [0u8; USIZE_LEN];
+	possiblyrandom::getpossiblyrandom(&mut random_bytes);
```

Comment:
It seems safe to assume that routing nodes have random available? Looking at possiblyrandom, isn't the same value picked all the time when `getrandom` isn't available?

Reply:
Isn't it better to control that fixed value (maybe 50ms always?) rather than taking the first from the list?

Reply:
I don't think that is necessary. An alternative could be to take the `getpossiblyrandom` code and instead of `dest.fill(0)`, return a fixed 50 ms from batchdelay?